src/main/java/com/altinity/clickhouse/sink/connector/db/DbWriter.java

```diff
@@ -623,42 +634,44 @@ public void insertPreparedStatement(Map<String, Integer> columnNameToIndexMap, P
 
         // Sign column.
         //String signColumn = this.config.getString(ClickHouseSinkConnectorConfigVariables.CLICKHOUSE_TABLE_SIGN_COLUMN);
-        if(this.engine != null && this.engine.getEngine() == DBMetadata.TABLE_ENGINE.COLLAPSING_MERGE_TREE.getEngine() &&
-        this.signColumn != null)
-        if (this.columnNameToDataTypeMap.containsKey(signColumn) && columnNameToIndexMap.containsKey(signColumn)) {
-            int signColumnIndex = columnNameToIndexMap.get(signColumn);
-            if (record.getCdcOperation().getOperation().equalsIgnoreCase(ClickHouseConverter.CDC_OPERATION.DELETE.getOperation())) {
-                ps.setInt(signColumnIndex, -1);
-            } else if (record.getCdcOperation().getOperation().equalsIgnoreCase(ClickHouseConverter.CDC_OPERATION.UPDATE.getOperation())){
-                if(beforeSection == true) {
-                    ps.setInt(signColumnIndex, - 1);
+        if(this.engine != null && this.signColumn != null &&
+            (this.engine.getEngine().equalsIgnoreCase(DBMetadata.TABLE_ENGINE.COLLAPSING_MERGE_TREE.getEngine()) ||
+            this.engine.getEngine().equalsIgnoreCase(DBMetadata.TABLE_ENGINE.REPLICATED_COLLAPSING_MERGE_TREE.getEngine()) ||
+            this.engine.getEngine().equalsIgnoreCase(DBMetadata.TABLE_ENGINE.REPLICATED_VERSIONED_COLLAPSING_MERGE_TREE.getEngine()) )){
+            if (this.columnNameToDataTypeMap.containsKey(signColumn) && columnNameToIndexMap.containsKey(signColumn)) {
+                int signColumnIndex = columnNameToIndexMap.get(signColumn);
+                if (record.getCdcOperation().getOperation().equalsIgnoreCase(ClickHouseConverter.CDC_OPERATION.DELETE.getOperation())) {
+                    ps.setInt(signColumnIndex, -1);
+                } else if (record.getCdcOperation().getOperation().equalsIgnoreCase(ClickHouseConverter.CDC_OPERATION.UPDATE.getOperation())){
+                    if(beforeSection == true) {
+                        ps.setInt(signColumnIndex, - 1);
```

Comment:
do we get before in case of incremental snapshot, in case the data is already present as part of CDC ? How is that handled ?